Documentation/custom_effects.md

```diff
@@ -33,23 +34,52 @@ To use a custom effect with MonoGame you must do one of the following (not both)
-* You can use `#if SM4` to add conditional code for DirectX platforms.
-* On DirectX platforms use the `SV_Position` semantic instead of `POSITION` in vertex shader inputs.
+* The supported shader models when targeting DX are the following:
+  * vs_4_0_level_9_1 and ps_4_0_level_9_1 
+  * vs_4_0_level_9_3 and ps_4_0_level_9_3
+  * vs_4_0 and ps_4_0 (requires HiDef GraphicsProfile at runtime)
+  * vs_4_1 and ps_4_1 (requires HiDef GraphicsProfile at runtime)
+  * vs_5_0 and ps_5_0 (requires HiDef GraphicsProfile at runtime)
+* When targeting GL platforms we automatically translate FX files to GLSL using a library called [MojoShader](http://icculus.org/mojoshader/).  The supported feature levels are the following:
+  * vs_2_0 and ps_2_0
+  * vs_3_0 and ps_3_0
+* You can use preprocessor checks to add conditional code or compilation depending on defined symbols. MonoGame defines the following symbols when compiling effects:
+  * `2MGFX`
+  * `HLSL` and `SM4` for DirectX
+  * `OpenGL` and `GLSL` for OpenGL
+  
+  As an example, you can conditionally set shader models depending on the platform with the following code:
+  ```
+  #if OPENGL
+      #define VS_SHADERMODEL vs_3_0
+      #define PS_SHADERMODEL ps_3_0
+  #else
+      #define VS_SHADERMODEL vs_4_0_level_9_1
+      #define PS_SHADERMODEL ps_4_0_level_9_1
+  #endif
+
+  technique
+  {
+      pass
+      {
+          VertexShader = compile VS_SHADERMODEL MainVS();
+          PixelShader = compile PS_SHADERMODEL MainPS();
+      }
+  };
+  ```
+  Custom symbols can be defined from the [Pipeline Tool](pipeline.md) or via [2MGFX](2mgfx.md).
+* Make sure the pixel shaders inputs **exactly match** the vertex shader outputs so the parameters are passed in the correct registers. The parameters need to have the same size and order. Omitting parameters might not break compilation, but can cause unexpected results.
 * Note that on GL platforms default values on Effect parameters do not work.  Either set the parameter from code or use a real constant like a #define.
-* Do not name your sampler `Sampler` - it will not compile.
-* The effect compiler is aggressive about removing unused paramters, be sure the parameters you are setting are actually used.
-* If you think you've found a bug porting a shader [please let us know](https://github.com/mono/MonoGame/issues).
+* The effect compiler is aggressive about removing unused parameters, be sure the parameters you are setting are actually used.
+* Preshaders are not supported.
+* If you think you've found a bug porting a shader [please let us know](https://github.com/MonoGame/MonoGame/issues).
 
 # Roadmap
 There is still work to be done for better support of custom effects and shaders in MonoGame:
 
 * Support GLSL in FX files.
   * Write a new preprocessor replacing [D3DPreprocess](http://msdn.microsoft.com/en-us/library/windows/desktop/dd607332.aspx).
```

Comment:
We need to remove the roadmap.  It is crazy out of date.

Reply:
Yeah, that's true. Just remove it altogether?

Reply:
Yeah... i don't think it serves any purpose here.

Reply:
Yeah, actually none of these are still planned.